content/blog/posts/2022-02-01-the-benefits-of-using-nextjs.md

```diff
@@ -0,0 +1,140 @@
+---
+authorId: JessicaJustice
+categories:
+  - next.js
+  - react
+  - javascript
+date: '2022-02-01'
+slug: when-to-use-nextjs
+title: 'The Benefits of Using Next.js'
+---
+
+## What Is Next.js?
+
+Next.js is a robust JavaScript framework created by [Vercel](https://vercel.com/) and designed for web application development. Importantly, Next.js builds upon React, a JavaScript library that makes building frontend user interfaces (UI) quick and efficient.
+
+In some instances, React is sufficient for building out the frontend of a successful web application on its own. However, many developers end up importing a number of external dependencies in order to meet their application's needs. Next.js, however, comes with a number of popular features already built in, while also providing functional and organizational benefits not found in React alone.
+
+## How Is Next.js Different From React?
+
+- React is a JavaScript library, while Next.js is considered a [React framework for production](https://nextjs.org/)
+- React is useful for building out a web application's frontend, while Next.js offers some full stack capabilities
+- React renders UI components on the client side of the application, while Next.js offers server-side rendering capabilities
+- React alone does not offer built-in page routing (although the [React Router package](https://reactrouter.com/) is available as a dependency), while Next.js features built-in file-based routing
+- As a framework, Next.js has a strict organizational structure that developers must follow, while React alone does not
+
+## How Does Next.js Build Upon React?
+
+Next.js retains the following features and capabilities offered by React:
+
+- Makes building reusable components relatively quick and easy
+- Only updates elements on the DOM after a relevant piece of state has been manipulated
+- Uses [JSX](https://reactjs.org/docs/introducing-jsx.html), a syntax that blends elements of JavaScript and HTML
+
+To build upon React, Next.js also offers additional features that help developers push their React applications into production faster. Below is an overview of some of those essential features that set Next.js apart from React.
+
+### Pre-Rendering Options
+
+On an initial load, web applications built with React will render a page's static elements on the client side. Then, the network requests necessary for fetching any relevant data for that page will fire. Any returned data will populate the page where appropriate.
+
+Depending on the number of network requests and the bulk of the data returned, this process can result in either a brief, or potentially sizable, loading time for the end user. To accomodate long loading times, developers often opt to include a helpful component to be displayed on the UI that indicates to the user that the page isn't frozen, just loading.
```

Comment:
Might be easier to understand from reversing this sentence:

"This process can result in a sizable loading time for the end user depending on the number of network requests and the [bulk of the] data required."

Also a bit confused on "the bulk of the." As in the amount of data?